packages/carbon-react/src/components/TextInput/TextInput.stories.js

```diff
@@ -0,0 +1,221 @@
+      ['start', 'center', 'end'],
+      'center'
+    ),
+    hidePasswordLabel: text(
+      '"Hide password" tooltip label for password visibility toggle (hidePasswordLabel)',
+      'Hide password'
+    ),
+    showPasswordLabel: text(
+      '"Show password" tooltip label for password visibility toggle (showPasswordLabel)',
+      'Show password'
+    ),
+  }),
+};
+
+TextInput.displayName = 'TextInput';
+
+export default {
+  title: 'Components/TextInput',
+  decorators: [withKnobs],
+
+  parameters: {
+    component: TextInput,
+    docs: {
+      page: mdx,
+    },
+    subcomponents: {
+      TextInputSkeleton,
+      'TextInput.PasswordInput': TextInput.PasswordInput,
+    },
+  },
+};
+
+export const Default = () => (
+  <TextInput
+    type={select('Form control type (type)', types, 'text')}
+    {...props.SharedInputProps()}
+    {...props.TextInputProps()}
+  />
+);
+
+Default.parameters = {
+  info: {
+    text: `
+        Text fields enable the user to interact with and input data. A single line
+        field is used when the input anticipated by the user is a single line of
+        text as opposed to a paragraph.
+        The default type is 'text' and its value can be either 'string' or 'number'.
+      `,
+  },
+};
+
+export const Fluid = () => (
+  <FluidForm>
+    <TextInput
+      type={select('Form control type (type)', types, 'text')}
+      {...props.SharedInputProps()}
+      {...props.TextInputProps()}
+    />
+  </FluidForm>
+);
+
+Fluid.parameters = {
+  info: {
+    text: `
+        Text fields enable the user to interact with and input data. A single line
+        field is used when the input anticipated by the user is a single line of
+        text as opposed to a paragraph.
+        The default type is 'text' and its value can be either 'string' or 'number'.
+      `,
+  },
+};
+
+export const TogglePasswordVisibility = () => {
+  return (
+    <TextInput.PasswordInput
+      {...props.SharedInputProps()}
+      {...props.PasswordInputProps()}
+    />
+  );
+};
+
+TogglePasswordVisibility.storyName = 'Toggle password visibility';
+
+TogglePasswordVisibility.parameters = {
+  info: {
+    text: `
+      Text field with password visibility toggle.
+    `,
+  },
+};
+
+export const FullyControlledTogglePasswordVisibility = () => {
+  ControlledPasswordInputApp.__docgenInfo = {
+    ...TextInput.PasswordInput.__docgenInfo,
+    props: {
+      ...TextInput.PasswordInput.__docgenInfo.props,
+    },
+  };
+
+  return (
+    <ControlledPasswordInputApp
+      {...props.SharedInputProps()}
+      {...props.PasswordInputProps()}
+    />
+  );
+};
+
+FullyControlledTogglePasswordVisibility.storyName =
+  'Fully controlled toggle password visibility';
+
+FullyControlledTogglePasswordVisibility.parameters = {
+  info: {
+    text: `
+    Fully controlled text field with password visibility toggle.
+  `,
+  },
+};
+
+export const Skeleton = () => <TextInputSkeleton />;
+
+Skeleton.storyName = 'skeleton';
```

Comment:
Storybook v6 automatically infers the name of the story from the export name, so I think we can remove these for most cases.